src-ghc/Pact/PersistPactDb/Regression.hs

```diff
@@ -85,19 +90,19 @@ runRegression p = do
 
     [ TxLog { _txDomain = "SYS_keysets"
             , _txKey = "ks1"
-            , _txValue = toJSON ks
+            , _txValue = toLegacyJson ks
```

Comment:
How will future readers know when and why a legacy function is needed, and when it can be lifted?

Reply:
Here the type checker will tell, because `_txValue` is of type `LegacyValue`. 

More generally `LegacyValue` is used only where required for backward compatible serialization. Otherwise, we really should stop misusing `Value` as a generic data representation. It is an internal aeson type and part of the API for writing JSON encoders and decoders. (And even aeson kind of deprecated it for encoding use.) It shouldn't be used for anything else.